### PR TITLE
Make installing the proxy from a repository archive the documented way

### PR DESCRIPTION
**Pull request by sgiehl**
## Why

The README documents `git clone` as one of the installation methods. A checkout contains a lot that an installation does not need — the test suite, the CI configuration, the coding standard rules and the git metadata itself — and none of that belongs in a public web root. Installing from an archive keeps the installed directory to the files the proxy actually consists of.

## What

- **`.gitattributes`**: `export-ignore` for `tests/`, `.github/`, `phpcs.xml`, `.editorconfig`, `.gitignore`, `.gitattributes`, `.php-version` and `composer.lock`. `git archive` output — and the source archives GitHub attaches to releases and serves from `/archive/` — then contain only `LICENSE`, `README.md`, `composer.json`, `config.php.example`, the four entry points and `plugins/HeatmapSessionRecording/configs.php`. `composer.json` is kept deliberately: dropping it would break Composer-based consumption and it only lists dependency names.
- **README**: the source archive becomes the documented installation method, with a `curl … | tar xz --strip-components=1` recipe and an update instruction noting that `config.php` is not part of the archive and is left untouched, while extracting over an existing installation never removes files. The clone instructions move to the contributing section as a development checkout, and the installation section points installations that contain a `tests/` directory at cleaning it up.
- **`tests/server/matomo.php`**: this stand-in for a Matomo server is only meaningful together with the test configuration, so it now runs only when the loaded config points at `tests/server/` and returns `403` otherwise — the same gate `.github/config.php` already uses for its test-only header hooks. Its non-image responses also get an explicit `text/plain` content type plus `nosniff` instead of falling back to whatever the web server assumes. A new test covers the refusal, via a new `X-Test-Non-Test-Config` hook in the test configuration.

## Notes

- `export-ignore` only affects archives generated after this is merged, so the archives of the existing tags still contain the development files. The README therefore points at the `master` archive, which is clean as soon as this lands. Once a tag is cut on top of this, the README can document the tagged archive as the primary download.
- The README's copy of the test `config.php` is kept in sync with `.github/config.php` by hand; that duplication predates this PR, but is worth replacing with a pointer at some point.

## Tests

`vendor/bin/phpunit -c tests/phpunit.xml.dist` — 95 passed, 254 assertions, including the new one. Verified that the new test fails when the check in `tests/server/matomo.php` is removed, and that `git archive` output matches the file list above. `vendor/bin/phpcs` is clean.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Note that this does NOT apply to `git clone` - a clone always contains everything, which is
+# why the README documents the release archive as the way to install the proxy.
+/tests          export-ignore
+/.github        export-ignore
+/.editorconfig  export-ignore
+/.gitattributes export-ignore
+/.gitignore     export-ignore
+/.php-version   export-ignore
+/composer.lock  export-ignore
+/phpcs.xml      export-ignore
```
Comment (tzi) [inline .gitattributes:14]: `export-ignore` is a denylist, so anything new at the top level lands in the archive by default. The next `psalm.xml` / `Makefile` / `docs/` added without a matching line here ships to every new install, and finding out requires listing an archive by hand.

Could we add a CI step asserting the archive contents, so this stays an invariant rather than a one-time cleanup?

Comparing `git archive HEAD | tar -t` against a checked-in expected list, or at minimum failing on a `tests/` or `.github/` entry:

```yaml
- name: Check archive contains no development files
  run: |
    git archive HEAD | tar -t | sort > /tmp/actual.txt
    diff -u .github/expected-archive-files.txt /tmp/actual.txt
```

Comment (sgiehl) [reply .gitattributes:14]: The file in this repo are changing so litte, that I don't consider it worth the effort for this PR. But feel free to create a follow-up PR if you consider this worth the effort.

Comment (tzi) [reply .gitattributes:14]: I do not know this repository very well. Thanks for the input.

**Review by tzi — APPROVED** [review]
